sled-agent/config-reconciler/src/reconciler_task.rs

```diff
@@ -127,21 +147,292 @@ pub enum ReconcilerTaskStatus {
 #[derive(Debug)]
 struct LatestReconcilerTaskResultInner {
     sled_config: OmicronSledConfig,
+    datasets: DatasetEnsureResult,
     timesync_status: TimeSyncStatus,
 }
 
-struct ReconcilerTask<T> {
+struct ReconcilerTask {
     key_requester: StorageKeyRequester,
-    time_sync_config: TimeSyncConfig,
+    dataset_task: DatasetTaskHandle,
     current_config_rx: watch::Receiver<CurrentSledConfig>,
     reconciler_result_tx: watch::Sender<ReconcilerResult>,
+    raw_disks_rx: RawDisksReceiver,
     external_disks: ExternalDisks,
-    sled_agent_facilities: T,
+    zones: OmicronZones,
     log: Logger,
 }
 
-impl<T: SledAgentFacilities> ReconcilerTask<T> {
-    async fn run(self) {
-        unimplemented!()
+impl ReconcilerTask {
+    async fn run<T: SledAgentFacilities>(mut self, sled_agent_facilities: T) {
+        // If reconciliation fails, we may want to retry it. The "happy path"
+        // that requires this is waiting for time sync: during RSS, cold boot,
+        // or replacement of the NTP zone, we may fail to start any zones that
+        // depend on time sync. We want to retry this pretty frequently: it's
+        // cheap if we haven't time sync'd yet, and we'd like to move on to
+        // starting zones as soon as we can.
```

Comment:
Makes sense to me; I'm not sure that backing off here would actually be useful for us.